pyteal/ast/gitxn.py

```diff
@@ -18,20 +19,23 @@ def __init__(self, txnIndex: int, field: TxnField) -> None:
         super().__init__(Op.gitxn, "Gitxn", field)
         self.txnIndex = txnIndex
 
+        def validate_types_or_throw():
+            # currently we do not have gitxns, only gitxn with immediate transaction index supported
+            if type(self.txnIndex) is not int:
+                raise TealInputError(
+                    "Invalid gitxn syntax with immediate transaction field {} and transaction index {}".format(
+                        self.field, self.txnIndex
+                    )
+                )
+
+        validate_types_or_throw()
```

Comment:
The logic look correct, but stylistically I would prefer not to define a function and immediately call it this way. If you want to abstract the logic of the check and make it into a helper function, that might make sense, but otherwise this seems more confusing than just directly executing the code